### DISCUSSION TITLE
Array format for bulk load COPY using psql

### DISCUSSION THREAD

**Discussion by shotleybuilder** [supabase/supabase#22629] [Questions]
Hi, really struggling to get the COPY bulk upload to accept an array with length > 1 (ie more than a single element array).
These work -> {'foobar'} || {"foobar"} || {foobar}
These do not -> {'foo','bar'} || {"foo","bar"} || {"foo", "bar"} or square brackets with those options
With a warning "extra data after last expected column".  The array column is the last one in my .csv file.  Works with a single element and doesn't with two.  Therefore, its not a problem with unmatched columns.

**Comment by GaryAustin1** [comment]
First hit I got on `psql csv array format`
https://briandfoy.github.io/importing-array-values-into-postgres-from-csv/
Not tested.

**Reply by shotleybuilder** [reply]
Thanks @GaryAustin1 I had seen that post.  I have tried pretty much every combo of format I can think of.  I even loaded a single element array into the column, did an update, and then read it back to see the formatting - that's where the square brackets came from.  I've done a little script that cleans up my data and i'm going to go back to trying the uploader again with the clean data.  All good material for a blog post :)

**Reply by shotleybuilder** [reply]
Blog https://spongl.ghost.io/bulk-upload-airtable-csv-supabase/
